### PR TITLE
Updated documentation on how to authenticate for the new Auth-api

### PR DESCRIPTION
**Pull request by PelleRavn**
We changed how to authentication works, where you previously authenticated to each API, but now moved to auth.mapsindoors.com. Updated the documentation to reflect how to do that.

### CODE REVIEW

**Comment by PelleRavn** [comment]
I would like a review on both the content, and the styling (this is why both Mark and Kasper is reviewers)

**Comment by PelleRavn** [comment]
Oh, just realizing, that it might already have been technical-reviewed in MIBAPI-912

File: collections/_api/v1/index.md [hunk 1]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-`Content-Type: application/x-www-form-urlencoded`
+```
+grant_type: google_id_token
+client_id: client
+id_token: <your Google token>
+```
 
-Use the following Key/Values
+The body of the request should end up containing a query string like this:
```
Comment (kblmapspeople) [inline collections/_api/v1/index.md]: Er det ikke formdata og ikke en query string?

Comment (kblmapspeople) [inline collections/_api/v1/index.md]: Jeg ved godt at det er i docen i dag men vil vi tilbyde google login på integration apiet? er det ikke mere en CMS ting?

Comment (PelleRavn) [reply collections/_api/v1/index.md]: Det er et godt spørgsmål. Umiddelbart tænker jeg at det lige så godt kan, da det jo i forvejen er centraliseret i Auth. Der skal vi måske have fat i @WJesper for et svar?

Comment (PelleRavn) [reply collections/_api/v1/index.md]: Jeg tjekkede, og det virker åbenbart med begge dele. 
Men tror det er langt nemmere for de fleste sprog at sende en query string, i stedet for formdata.

File: collections/_api/v1/index.md [hunk 2]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
 
 From the MapsIndoors Integration API you can get, add, change and delete data related to your MapsIndoors™ solution via a REST service.
 
-The Integration API can be found here: [https://integration.mapsindoors.com](https://integration.mapsindoors.com)
-
-There is a (Swagger) interface definition here: [https://integration.mapsindoors.com/doc](https://integration.mapsindoors.com/doc)
+Requests to the Integration API should be made to the following endpoint: [https://integration.mapsindoors.com](https://integration.mapsindoors.com)
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Tighten it up a bit:

> Send your requests to this endpoint: [url]

File: collections/_api/v1/index.md [hunk 3]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
 > Note: Only https is supported.
 {: .mi-careful}
 
-### Login and credentials
+The Integration API consists of a number of endpoints to access the various data. The description of those can be found in the Swagger interface definition available here: [https://integration.mapsindoors.com/doc](https://integration.mapsindoors.com/doc)
```
Comment (marksdk) [inline collections/_api/v1/index.md]: > You can access data through the Integration API using a range of endpoints. The endpoints are described in the Swagger interface definition: (url)

File: collections/_api/v1/index.md [hunk 4]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-The first thing needed is to log in to the service and get a security token used for the data endpoints.
+In Swagger, each GET method are pre-loaded with all mandatory fields needed for you to get a live example of data just by pressing the "_Try it out_" button.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: > In Swagger, each `GET` method is pre-loaded with all mandatory fields needed to get a live example of data. Click  the "_Try it out_" button in Swagger to see the example data.

File: collections/_api/v1/index.md [hunk 5]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-[HttpPost, Route("token")]
+### Login and credentials
 
-The interface supports two ways to login. Via Google or via a simple username/password pair.
+The first thing needed is to log in to the service and get a security token used for the data endpoints.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: What does "the service" here refer to?

I'd shorten this a bit to read:

> First, log in to the Swagger service to get a security token to access the data.

File: collections/_api/v1/index.md [hunk 6]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-**Login with username/password**
+This requires a POST request to our Auth API at the following endpoint: https://auth.mapsindoors.com/connect/token
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Is this relevant for the person reading the docs going through Swagger? Or only for someone requesting from elsewhere? If so, we should make that more clear.

File: collections/_api/v1/index.md [hunk 7]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-Use the following headers:
+The interface supports multiple ways to log in, but mostly commonly is via Google or a simple username/password pair.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: > Swagger supports multiple ways to log in. The most common way is authenticating with Google, but you can also use your MapsIndoors username and password.

File: collections/_api/v1/index.md [hunk 8]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
+No matter what login method you choose, you will always need to use the following content-type header when talking to the Auth API:
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Line-break before this line. Markdown doesn't do well when to separate lines don't have an empty one between them.

File: collections/_api/v1/index.md (hunk 7 above)
Comment (marksdk) [reply collections/_api/v1/index.md]: Is the above correct? I find it a bit confusing to be honest.

File: collections/_api/v1/index.md [hunk 9]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
 ```
 Content-Type: application/x-www-form-urlencoded
 ```
 
-Use the following Key/Values
+#### Login with MapsIndoors™ username/password
```
Comment (marksdk) [inline collections/_api/v1/index.md]: 1. "Login" -> "Log in"
1. MapsIndoors is not trademarked 🙂
1. "username/password" -> "username and password"

File: collections/_api/v1/index.md [hunk 10]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
+
+To sign in with your MapsIndoors™ login, you will need to send them with the `grant_type` set to `password`.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: > To log in with your MapsIndoors login, send them with [...]

File: collections/_api/v1/index.md [hunk 11]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
-Key: username Value: <your username>
-Key: password Value: <your password>
+grant_type: password
+client_id: client
+username: <your username>
+password: <your password>
 ```
 
-The body of the request should end up containing a text like this:
+The body of the request should end up containing a query string like this:
```
Comment (marksdk) [inline collections/_api/v1/index.md]: "Should" or "must"?

File: collections/_api/v1/index.md [hunk 12]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
+`grant_type=password&client_id=client&username=<your username>&password=<your password>`
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Line-break above here.

File: collections/_api/v1/index.md [hunk 13]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-`grant_type=password&username=<your username>&password=<your password>`
+#### Login with Google
```
Comment (marksdk) [inline collections/_api/v1/index.md]: "Login" -> "Log in"

File: collections/_api/v1/index.md [hunk 14]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
 
-To use Google login you will need to set `grant_type` to `google_id_token` instead.
+The [Google Accounts API](https://developers.google.com/identity/protocols/OAuth2) is to be used to obtain a valid Google token. Explaining this in detail is outside the scope of this document.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: "is to be used" -> "is used".

"Explaining this in detail is outside the scope of this document." -> "You can read more about how to do that here."

^ Let's help the reader a bit with a link to the relevant documentation for obtaining that token.

File: collections/_api/v1/index.md [hunk 15]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
+`grant_type=password&client_id=client&username=<your username>&password=<your password>`
 
-`grant_type=password&username=<your username>&password=<your password>`
+#### Login with Google
 
-To use Google login you will need to set `grant_type` to `google_id_token` instead.
+The [Google Accounts API](https://developers.google.com/identity/protocols/OAuth2) is to be used to obtain a valid Google token. Explaining this in detail is outside the scope of this document. 
 
-The [Google Accounts API ](https://developers.google.com/identity/protocols/OAuth2)is to be used to obtain a valid Google token. This is outside the scope of this document. 
+When you got a valid response back from the Google Authorization Server, you need to send the token to the Auth API to authenticate you.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: > When you get a valid response from the Google Authorization Server, send the token to the Auth API to authenticate you.

File: collections/_api/v1/index.md [hunk 16]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
-`Content-Type: application/x-www-form-urlencoded`
+```
+grant_type: google_id_token
+client_id: client
+id_token: <your Google token>
+```
 
-Use the following Key/Values
+The body of the request should end up containing a query string like this:
+`grant_type=google_id_token&client_id=client&id_token=<your Google token>`
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Line break above.

File: collections/_api/v1/index.md [hunk 17]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
+
+#### When you are authenticated
+
+If you have sent valid credentials to the Auth API, you will receive a response looking like the following:
```
Comment (marksdk) [inline collections/_api/v1/index.md]: > If you sent valid credentials to the Auth API, you will receive a response like this:

File: collections/_api/v1/index.md [hunk 18]
```diff
@@ -12,63 +12,84 @@ date: 2019-09-30
...
-### Endpoints
+You will need the value from the key `access_token` for all your requests to the Integration API by adding the `Authorization` header like this:
 
-The Integration API consists of a number of endpoints to access the various data. The description of those can be found in [Swagger](https://integration.mapsindoors.com/doc).
+```
+authorization: Bearer eyJhbGciOiJ...vmERrovsg
+```
 
-In Swagger each GET method are pre-loaded with all mandatory fields needed for you to get a live example of data just by pressing the "_Try it now_" button.
+> Note: The access token is valid for 24 hours. After that you will need to reauthenticate, following the same steps as explained above.
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Line break above.

File: collections/_api/v1/index.md [hunk 19]
```diff
@@ -482,7 +503,7 @@ Common setup for Geodata of different kinds (meeting room, hallway, ...)
 
     The name property must be specified for every language defined in the dataset.
 
-> Note: The display type data is split in the CMS; currently it can be found in the fans "Location types", "Location type templates" and "Type visibility".
+> Note: The display type data is split in the CMS; currently it can be found in the fans "Location Types" and "Type Visibility".
```
Comment (marksdk) [inline collections/_api/v1/index.md]: Kudos on the semi-colon; also, this was a good catch 😉

File: collections/_api/v1/index.md (hunk 6 above)
Comment (PelleRavn) [reply collections/_api/v1/index.md]: It's actually both. Swagger just does it for you if trying it out though Swagger. When you need your application to do this, you need to call this endpoint programmically. So I think it is phrased correctly as a general explanation on how to authenticate.

File: collections/_api/v1/index.md (hunk 7 above)
Comment (PelleRavn) [reply collections/_api/v1/index.md]: It's not completely correct. It's not Swagger that supports logging in like that. It's the Auth API/Integration API that supports it, and Swagger is just an easy-to-understand interface, and really couldn't be mentioned anymore, than just "There is a Swagger interface available".

Comment (marksdk) [reply collections/_api/v1/index.md]: @PelleRavn Alrighty.

**Comment by marksdk** [comment]
Good job @PelleRavn 👍
